python/tests/geopandas/test_sjoin_match.py

```diff
@@ -0,0 +1,43 @@
+import pytest
+import geopandas as gpd
+from shapely.geometry import Point
+from sedona.geopandas import sjoin
```

Comment:
```suggestion
from sedona.spark.geopandas import sjoin
```

I tried running this locally and found that there are also multiple issues with this test code (more than just this), that prevent it from running properly. Simple things like data types issues. @bunnysocks Do you need help with setting up your developer environment? You can follow [these directions](https://github.com/apache/sedona/blob/293faffd958af937a7c01b8f32da86c556d7e425/docs/setup/compile.md#L4) for setting it up, so you can test locally. Getting it working by following those instructions should work pretty smoothly. The logic this PR proposes is on the right track, but it will be very helpful for you to run the tests locally to avoid simple errors. Feel free to ping if you get stuck or need help, including with dev env setup.

e.g failed CI run
https://github.com/apache/sedona/actions/runs/18967529015/job/54243890091?pr=2466